src/menu.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib_ffi;
+use glib::object::Downcast;
+use glib::translate::*;
+use Menu;
+use IsA;
+use Widget;
+use std::boxed::Box as Box_;
+use std::mem::transmute;
+use libc::c_int;
+use std::ptr;
+
+impl Menu {
+    pub fn popup<T: IsA<Widget>, U: IsA<Widget>,
+                 F: Fn(&Menu, &mut i32, &mut i32, &mut bool) + 'static>(
```

Comment:
I'm sorry for the runaround with `x` and `y` but the boolean definitely is an outparam making it the return value.